ibm/service/usagereports/resource_ibm_billing_report_snapshot.go

```diff
@@ -0,0 +1,475 @@
+				ValidateFunc: validate.InvokeValidator("ibm_billing_report_snapshot", "interval"),
+				Description:  "Frequency of taking the snapshot of the billing reports.",
+			},
+			"versioning": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Default:      "new",
+				ValidateFunc: validate.InvokeValidator("ibm_billing_report_snapshot", "versioning"),
+				Description:  "A new version of report is created or the existing report version is overwritten with every update.",
+			},
+			"report_types": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: "The type of billing reports to take snapshot of. Possible values are [account_summary, enterprise_summary, account_resource_instance_usage].",
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+			"cos_reports_folder": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "IBMCloud-Billing-Reports",
+				Description: "The billing reports root folder to store the billing reports snapshots. Defaults to \"IBMCloud-Billing-Reports\".",
+			},
+			"cos_bucket": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The name of the COS bucket to store the snapshot of the billing reports.",
+			},
+			"cos_location": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Region of the COS instance.",
+			},
+			"state": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Status of the billing snapshot configuration. Possible values are [enabled, disabled].",
+			},
+			"account_type": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Type of account. Possible values are [enterprise, account].",
+			},
+			"compression": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Compression format of the snapshot report.",
+			},
+			"content_type": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Type of content stored in snapshot report.",
+			},
+			"cos_endpoint": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The endpoint of the COS instance.",
+			},
+			"created_at": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: "Timestamp in milliseconds when the snapshot configuration was created.",
+			},
+			"last_updated_at": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: "Timestamp in milliseconds when the snapshot configuration was last updated.",
+			},
+			"history": {
+				Type:        schema.TypeList,
+				Computed:    true,
+				Description: "List of previous versions of the snapshot configurations.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"start_time": {
+							Type:        schema.TypeInt,
+							Optional:    true,
+							Description: "Timestamp in milliseconds when the snapshot configuration was created.",
```

Comment:
All child arguments of a computed field should be computed. It cannot be Optional.

Reply:
Done.